src/main/java/org/sagebionetworks/bridge/hibernate/HibernateSurveyDao.java

```diff
@@ -0,0 +1,365 @@
+            
+            SurveyId surveyId = new SurveyId(keys);
+            hibernateHelper.deleteById(HibernateSurvey.class, surveyId);
+            
+            // Delete the schemas as well, or they accumulate.
+            try {
+                StudyIdentifier studyId = new StudyIdentifierImpl(existing.getStudyIdentifier());
+                uploadSchemaService.deleteUploadSchemaByIdPermanently(studyId, existing.getIdentifier());
+            } catch(EntityNotFoundException e) {
+                // This is OK. Just means this survey wasn't published.
+            }
+        }
+    }
+    
+    @Override
+    public Survey getSurvey(StudyIdentifier studyIdentifier, GuidCreatedOnVersionHolder keys, boolean includeElements) {
+        if (keys.getGuid().toLowerCase().startsWith(IDENTIFIER_PREFIX)) {
+            ImmutableMap<String,Object> params = ImmutableMap.of(
+                    "studyKey", studyIdentifier.getIdentifier(),
+                    "identifier", keys.getGuid().substring(IDENTIFIER_PREFIX.length()),
+                    "createdOn", keys.getCreatedOn());
+            
+            String getQuery = SELECT_TEMPLATE + GET_ACTIVE_WITH_IDENTIFIER;
+            
+            List<HibernateSurvey> results = hibernateHelper.queryGet(
+                    getQuery, params, OFFSET, PAGESIZE, HibernateSurvey.class);
+            
+            if (!results.isEmpty()) {
+                return results.get(0);
+            }
+            return null;
+        }
+        SurveyId surveyId = new SurveyId(keys);
+        Survey ret = hibernateHelper.getById(HibernateSurvey.class, surveyId);
+        if (ret != null && includeElements) {
+            attachSurveyElements(ret);
+        }
+        return ret;
+    }
+
+    @Override
+    public String getSurveyGuidForIdentifier(StudyIdentifier studyIdentifier, String surveyId) {
+        ImmutableMap<String, Object> params = ImmutableMap.of(
+                "studyKey", studyIdentifier.getIdentifier(),
+                "identifier", surveyId);
+        
+        String getQuery = SELECT_TEMPLATE + GET_ACTIVE_WITH_IDENTIFIER;
+        
+        List<? extends HibernateSurvey> results = hibernateHelper.queryGet(
+                getQuery, params, OFFSET, PAGESIZE, HibernateSurvey.class);
+        if (!results.isEmpty()) {
+            return results.get(0).getGuid();
+        }
+        return null;
+    }
+
+    @Override
+    public List<Survey> getSurveyAllVersions(StudyIdentifier studyIdentifier, String guid, boolean includeDeleted) {
+        ImmutableMap<String, Object> params = ImmutableMap.of(
+                "studyKey", studyIdentifier.getIdentifier(),
+                "guid", guid);
+        
+        String getQuery = SELECT_TEMPLATE + ((!includeDeleted) ? GET_ACTIVE : GET_ALL);
+        
+        List<HibernateSurvey> results = hibernateHelper.queryGet(
+                getQuery, params, OFFSET, 50, HibernateSurvey.class);
```

Comment:
The DynamoSurveyDao implementation didn't specify a page size, even though it probably should have. However, because it didn't, we don't have a pagination API for surveys, so we need to return all of them. HibernateHelper.queryGet() can take in null for the page size, meaning no page size.